GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -140,5 +141,41 @@ public override bool IsGitStatusCacheSupported()
         {
             return new MacFileBasedLock(fileSystem, tracer, lockPath);
         }
+
+        public override bool OpenAndReadFile(string fileName)
+        {
+            return NativeFileReader.OpenAndReadFile(fileName);
+        }
+
+        private class NativeFileReader
+        {
+            // #define O_RDONLY     0x0000      
+            public const int ReadOnly = 0x0000;
+
+            public static bool OpenAndReadFile(string fileName)
+            {
+                int fileDescriptor = Open(fileName, ReadOnly);
+                int success = ReadOneByte(fileDescriptor);
+
+                if (success == -1)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            private static unsafe int ReadOneByte(int fileDescriptor)

```

Comment:
Maybe make this `TryReadOneByte` and have it return a bool?

It looked a bit odd to see:

```int success = ReadOneByte(fileDescriptor);```

When `ReadOneByte` is returning the number of bytes read.